components/brave_rewards/resources/ui/components/checkout/addFundsPanel/index.tsx

```diff
@@ -0,0 +1,175 @@
+                  size={'medium'}
+                  onClick={selectAmount}
+                />
+                <AmountOptionExchange>
+                  {option.amountConverted}
+                </AmountOptionExchange>
+              </AmountOptionContainer>
+            )
+          })
+        }
+      </AmountOptionList>
+      <ChargeSummary>
+        <div>{locale.get('transactionFee')} ({selectedOption.transactionFeeRate})</div>
+        <div>{selectedOption.transactionFee}</div>
+        <ChargeSummaryTotal>{locale.get('orderTotal')}</ChargeSummaryTotal>
+        <ChargeSummaryTotalAmount>{selectedOption.totalCharge}</ChargeSummaryTotalAmount>
+      </ChargeSummary>
+    </>
+  )
+}
+
+interface AddFundsPanelProps {
+  amountNeeded: string
+  walletBalance: string
+  walletBalanceConverted: string
+  unitValueConverted: string
+  amountOptions: AddFundsAmountOption[]
+  onCancel: () => void
+  onPayWithCreditCard: (cardDetails: CreditCardDetails) => void
+}
+
+export function AddFundsPanel (props: AddFundsPanelProps) {
+  const locale = React.useContext(LocaleContext)
+
+  const [selectedAmount, setSelectedAmount] = React.useState<number>(0)
+  const creditCardFormRef = React.useRef<CreditCardFormHandle>(null)
+
+  const onPurchaseClick = () => {
+    const formHandle = creditCardFormRef.current
+    if (formHandle) {
+      const errors = formHandle.validate()
+      if (errors.length === 0) {
+        props.onPayWithCreditCard(formHandle.details)
+      } else {
+        errors[0].element.focus()
+      }
+    }
+  }
+
+  return (
+    <>
+      <h1>{locale.get('addFundsTitle')}</h1>
+      <Subtitle>
+        {locale.get('addFundsSubtitle')}
+      </Subtitle>
+      <CurrentBalance>
+        <div>
+          {locale.get('currentBalance')}
+          <CurrentBalanceBat>{props.walletBalance} {locale.get('bat')}</CurrentBalanceBat>
+          <CurrentBalanceConverted>{props.walletBalanceConverted}</CurrentBalanceConverted>
+        </div>
+        <CurrentBalanceNeeded>
+          {props.amountNeeded} {locale.get('batNeeded')}
+        </CurrentBalanceNeeded>
+      </CurrentBalance>
+      <FormSection
+        title={
+          <>
+            <div>1. ${locale.get('selectAmountToAdd')}</div>
+            <div>
+              <ExchangeRateDisplay>
+                <BatColorIcon /> 1 {locale.get('bat')} = {props.unitValueConverted}
+              </ExchangeRateDisplay>
+            </div>
+          </>
+        }
+      >
+        <AmountOptionPanel
+          amountOptions={props.amountOptions}
+          selectedAmount={selectedAmount}
+          setSelectedAmount={setSelectedAmount}
+        />
+      </FormSection>
+      <FormSection title={`2. ${locale.get('enterCreditCardInfo')}`}>
+        <CreditCardForm handleRef={creditCardFormRef} />
+      </FormSection>
+      <PurchaseButtonRow>
+        <div>
+          <GoBackLink onClick={props.onCancel} />
+        </div>
+        <div>
+          <Button
```

Comment:
I see that we have design for `PurchaseButtonRow button`. Let's try to use approach if possible. Name this one something like `AddFundsButton`. Then in style file you can do this

```
 export const PurchaseButtonRow = styled(Button as ComponentType<ButtonProps>)`
    padding-left: 42px;
    padding-right: 42px;
`

```

This way we remove nesting in `style.ts`